test/test_nn.py

```diff
@@ -10116,6 +10116,59 @@ def test_grid_sample_3d_alert_nondeterministic(self, device):
         output = F.grid_sample(input, grid, align_corners=False)
         output.sum().backward()
 
+    @dtypes(torch.float, torch.double)
+    @largeTensorTest(lambda self, device, dtype:
+                     32769 * (65536 + 3 * 65536 / 128) *
+                     torch.tensor([], dtype=dtype).element_size())
+    def test_grid_sample_large_index_2d(self, device, dtype):
```

Comment:
It would be really nice if we could have a test case that didn't have the @largeTensorTest annotation. If we, for example, accidentally uninstall psutil from all of our CI images, then all of the large tensor tests would stop running.  I haven't worked out the math for this so this might not actually be physically possible.

Reply:
The problem is that it's not just the byte offset that has to exceed 32-bits (in which case 2GB would be enough), it's actually the pointer offset in units of element size. So, 2^31 * 4 bytes = 8 GB for `float` and 2^31 * 8 bytes = 16 GB for `double`. That explains why nobody has run into this on cuda.

For what it's worth, I have access to a quadro card with 48 GB of memory and these tests do all run and pass.

Reply:
I see, that makes more sense now